### PR TITLE
Improve front-End validation on the checkout page

### PR DESCRIPTION
**Pull request by irenenguyen1017**
I've made several updates in this PR that I'd like to share:

**Front-End Data Validation**: 
- Improved validation for the shipping address on the checkout page to enhance user experience. In fact, this solution is only temporary. It merely instructs the user not to leave the field blank. However, there are still some unresolved issues. For instance, users can still enter a string as a postcode and proceed with an order, which doesn't make sense. I need to move on to testing now, but I'll address this weakness during the presentation to the educator.

Screenshot 2023-12-21 at 6 59 22 pm

Screenshot 2023-12-21 at 7 04 22 pm

**Shipping Address Visibility**: 
- Adjusted the shipping address accordion to be visible immediately upon navigating to the checkout page. This change aims to reduce user confusion and ensure a smoother checkout process, as previously, the accordion was hidden and might have deterred some users from completing their purchase.

Before
Screenshot 2023-12-21 at 7 36 08 pm

After
Screenshot 2023-12-21 at 7 36 47 pm

**Code Refactoring**: 
- Conducted some refactoring to improve code readability and maintainability.

**Post-Checkout Redirection**: 
- Modified the flow to redirect users to the order success page after completing their checkout. This addresses an issue where users could still access the checkout page post-purchase.

**Product Detail Enhancements:** 
- Set the default quantity to 1 for a better initial user experience, ensuring users start with a single item in their cart.
- Tweaked the layout of the product image to centralise it, enhancing the visual appeal.

Screenshot 2023-12-21 at 7 37 33 pm

There are a few issues that still need improvement, and they might take some time to address. For now, I'm going to leave them as they are and revisit them in the future. One small thing to note: the border seems to be missing a part when a user presses the tab key. It's not a major issue, just something I wanted to mention for future reference.

Screenshot 2023-12-21 at 7 39 58 pm

### CODE REVIEW

File: src/components/Checkout/Checkout.jsx [hunk 1]
```diff
@@ -106,162 +105,156 @@ export const Checkout = () => {
...
-      console.log(updatedData);
       await placeOrder(updatedData);
     } catch (error) {
-      console.error('Error submitting form:', error);
       toast.error('An error occurred while placing the order.');
     }
   };
 
+  // If the cart is empty, redirect to the homepage.
+  if (!cartItems?.length) {
```
Comment (irenenguyen1017) [inline src/components/Checkout/Checkout.jsx:115]: We should implement this check to block scenarios where users attempt to access the checkout page without any items in their cart and try to place orders.

File: src/components/Checkout/Checkout.jsx [hunk 2]
```diff
@@ -106,162 +105,156 @@ export const Checkout = () => {
...
+    return <Navigate replace to={'/'} />;
+  }
+
+  // If order is placed sucessfully, it should show success screen
+  if (orderPlaced) {
```
Comment (irenenguyen1017) [inline src/components/Checkout/Checkout.jsx:120]: I moved this code here to make it easy to read.

**Review by Mana12011207 — APPROVED** [review]

**Review by ellenpham — APPROVED** [review]

File: src/components/Checkout/Checkout.jsx (hunk 1 above)
Comment (ellenpham) [reply src/components/Checkout/Checkout.jsx:114]: all good

File: src/components/Checkout/Checkout.jsx (hunk 2 above)
Comment (ellenpham) [reply src/components/Checkout/Checkout.jsx]: sure, thank you

**Pull request merged**